nemo/export/quantize/quantizer.py

```diff
@@ -72,89 +79,13 @@ class Quantizer:
     for TensorRT-LLM deployment. This is useful to getting baseline results for a full-precision model.
     """
 
-    def __init__(
-        self,
-        quantization_config: DictConfig,
-        inference_config: DictConfig,
-        export_config: DictConfig,
-        trainer_config: DictConfig,
-    ):
-        if not HAVE_MODELOPT:
-            raise RuntimeError("nvidia-modelopt is needed to use Quantizer") from HAVE_MODELOPT_ERROR
-        QUANT_CFG_CHOICES = {
-            "int8": mtq.INT8_DEFAULT_CFG,
-            "int8_sq": mtq.INT8_SMOOTHQUANT_CFG,
-            "fp8": mtq.FP8_DEFAULT_CFG,
-            "int4_awq": mtq.INT4_AWQ_CFG,
-            "w4a8_awq": mtq.W4A8_AWQ_BETA_CFG,
-        }
-        SUPPORTED_DTYPE = [16, "16", "bf16"]  # Default precision for non-quantized layers
-        assert export_config.dtype in SUPPORTED_DTYPE
-        assert quantization_config.algorithm is None or quantization_config.algorithm in QUANT_CFG_CHOICES
-        self.quantization_config = quantization_config
-        self.inference_config = inference_config
-        self.export_config = export_config
-        self.trainer_config = trainer_config
-        if quantization_config.algorithm is not None:
-            quant_cfg = QUANT_CFG_CHOICES[quantization_config.algorithm]
-
-            if "awq" in quantization_config.algorithm:
-                weight_quantizer = quant_cfg["quant_cfg"]["*weight_quantizer"]
-                if isinstance(weight_quantizer, list):
-                    weight_quantizer = weight_quantizer[0]
-                weight_quantizer["block_sizes"][-1] = quantization_config.awq_block_size
-
-            # Always turn on FP8 kv cache to save memory footprint.
-            # For int8_sq, we use int8 kv cache.
-            # TODO: Investigate why enabling FP8 kv cache will cause accuracy regressions for Nemotron.
-            enable_quant_kv_cache = (
-                "int8" not in quantization_config.algorithm and export_config.decoder_type != "gptnext"
-            )
-            print(f'{"Enable" if enable_quant_kv_cache else "Disable"} KV cache quantization')
-            quant_cfg["quant_cfg"]["*output_quantizer"] = {
-                "num_bits": 8 if quantization_config.algorithm == "int8_sq" else (4, 3),
-                "axis": None,
-                "enable": enable_quant_kv_cache,
-            }
-
-            self.quant_cfg = quant_cfg
-        else:
-            self.quant_cfg = None
-
-    def _load_model(
```

Comment:
model/trainer loading may be different for different pipelines (e.g. SFT uses a different model and trainer class and has some other changes) so moved this part to the example script instead of this common place

Reply:
OK